test/ducks/apps/components/applicationPage/details.spec.js

```diff
@@ -67,4 +73,101 @@ describe('ApplicationPage details component', () => {
     const component = shallow(<Details {...appProps} />).getElement()
     expect(component).toMatchSnapshot()
   })
+
+  it('should handle channel switching if with installed app from registry provided', async () => {
+    jest.doMock('ducks/apps', () => ({
+      REGISTRY_CHANNELS,
+      getContext: () =>
+        Promise.resolve({
+          attributes: {
+            debug: true
+          }
+        })
+    }))
+    // we have to re import to handle previous mocking here
+    const MockedDetails = require('ducks/apps/components/ApplicationPage/Details')
+      .Details
+    const appProps = getAppProps()
+    // correctly handled if from registry and installed
+    appProps.app.installed = true
+    appProps.app.source = `registry://${appProps}/stable`
+    const component = shallow(<MockedDetails {...appProps} />)
+    expect(component.state('displayBetaChannel')).toBe(false)
+    expect(component.state('displayDevChannel')).toBe(false)
+    await component.instance().toggleChannels()
+    expect(component.state('displayBetaChannel')).toBe(true)
+    expect(component.state('displayDevChannel')).toBe(true)
+    component.update()
+    expect(component.getElement()).toMatchSnapshot()
+  })
+
+  it('should not suggest Dev channel switching if the context is not in debug mode', async () => {
+    jest.doMock('ducks/apps', () => ({
+      REGISTRY_CHANNELS,
+      getContext: () =>
+        Promise.resolve({
+          attributes: {
+            debug: false
+          }
+        })
+    }))
+    // we have to re import to handle previous mocking here
+    const MockedDetails = require('ducks/apps/components/ApplicationPage/Details')
+      .Details
+    const appProps = getAppProps()
+    // correctly handled if from registry and installed
+    appProps.app.installed = true
+    appProps.app.source = `registry://${appProps}/stable`
+    const component = shallow(<MockedDetails {...appProps} />)
+    expect(component.state('displayBetaChannel')).toBe(false)
+    expect(component.state('displayDevChannel')).toBe(false)
+    await component.instance().toggleChannels()
+    expect(component.state('displayBetaChannel')).toBe(true)
+    expect(component.state('displayDevChannel')).toBe(false)
+    component.update()
+    expect(component.getElement()).toMatchSnapshot()
+  })
+
+  it('should not suggest channel switching if the app is not from registry or not installed', async () => {
+    jest.doMock('ducks/apps', () => ({
+      REGISTRY_CHANNELS,
+      getContext: () =>
+        Promise.resolve({
+          attributes: {
+            debug: true
+          }
+        })
+    }))
+    // we have to re import to handle previous mocking here
+    const MockedDetails = require('ducks/apps/components/ApplicationPage/Details')
+      .Details
+    const appProps = getAppProps()
+    // app not installed
```

Comment:
Every one of these three cases should be an unique test, that's the role of an _unit_ test : test only one condition and expected result.

Same remark about testing the implementation.